stdlib/camlinternalFormat.ml

```diff
@@ -464,16 +464,16 @@ let string_of_formatting_lit formatting_lit = match formatting_lit with
   | Flush_newline        -> "@."
   | Magic_size (str, _)  -> str
   | Escaped_at           -> "@@"
-  | Escaped_percent      -> "@%"
+  | Escaped_percent      -> "@%%"
```

Comment:
I missed the fact that `string_of_formatting_lit` is shared with `printf` implementation, and thus it must output `"@%"` here since `Printf.sprintf "@%%"` should be `@%` (due to `@` not being a formatting character in `Printf`).

Reply:
Indeed, you are correct. I propose to keep `string_of_formatting_lit` unchanged (keep the PR change to `string_of_formatting_gen` which makes it more consistent with `string_of_formatting_lit`), and to stop using them from the internal format printer, using special-purpose functions that only perform escaping in the right places. Does that sound like a sensible approach to you?